### PR TITLE
feature: add digital signature on the OEM

### PR DESCRIPTION
**Pull request by mihai1voicescu**
To be moreged before https://github.com/Almer-Technologies/flasher/pull/147

### CODE REVIEW

File: src/serial_number.ts [hunk 1]
```diff
@@ -1,27 +1,51 @@
...
+export const createImageFile = async (serialNumber: string, signature: string) => {
     if (!isValidSerialNumber(serialNumber)) {
         throw new Error("Invalid serial number");
     }
 
-    // Create a 1024KB blob filled with zeros
-    const size = 1024 * 1024; // 1024KB
-    const zeroFilledArray = new Uint8Array(size).fill(0);
-    let blob = new Blob([zeroFilledArray]);
+    if (typeof signature !== "string") {
```
Comment (niklas-pernter) [inline src/serial_number.ts:26]: can there be an incorrect signature as well?

Comment (mihai1voicescu) [reply src/serial_number.ts:26]: Yes, but verifying the signature is out of scope. You need the public key to verify this.

Comment (mihai1voicescu) [reply src/serial_number.ts:26]: The flasher generates the signature and also offers a validation endpoint for the public key.

Comment (niklas-pernter) [reply src/serial_number.ts:26]: Means the flasher only creates the valid signatures? 

can you elaborate on the signature creation so we/you can implement it in the Flasher

Comment (niklas-pernter) [reply src/serial_number.ts:26]: Ah you explained it before i asked hahah

Comment (niklas-pernter) [reply src/serial_number.ts:26]: you already implemented it xDD. You're fucking fast hahahaa

**Review by Taving40 — APPROVED** [review]

File: src/serial_number.ts (hunk 1 above)
Comment (mihai1voicescu) [reply src/serial_number.ts:26]: 😸

**Pull request merged**